tree/tree/src/TBranch.cxx

```diff
@@ -1279,6 +1279,14 @@ Int_t TBranch::GetEntry(Long64_t entry, Int_t getall)
             fNextBasketEntry = -1;
             return -1;
          }
+         if (fTree->GetMaxVirtualSize() < 0) {
```

Comment:
> makes sense if one also drops clusters, rather than baskets.

The current semantic of GetMaxVirtualSize is to indicates how much basket the TTree can keep in memory and not some much the granularity at which they are unloaded from memory.

> the intention was to make GetMaxVirtualSize similar to TTreeCache size, where a positive value indicates the number of bytes and a negative value uses clusters.

That makes sense only if the any (integer) negative value is 'properly supported' so that the user can give the TTree an arbitrary amount of memory.

The downside is that in this mode the memory used in unpredictable (there might be a stray 'very' large cluster every once in a while).

So one could also consider to keep GetMaxVirtualSize as it is and add a new concept: "granularity at which baskets are loaded and unloaded from memory" (in which case the memory would still be a bit unpredictable (since there would be at least one whole uncompressed cluster in memory) but less so (the max memory used would max(sizeof(one cluster), user_given_memory_barrier)